wunderio/next-drupal-starterkit
# Updates, various small improvements to configs etc

## Link to feature environment:

https://feature-updates-etc-again-n53e.next-drupal-starterkit.dev.wdr.io

## Changes proposed in this PR:

- Update lando cypress node image from node 20.11.0 to 20.18.0 (before this update, I was seeing an error with latest lando v3.23.22)
- Update dependencies to latest versions where possible (Next.js is still on v14 until @neshca/cache-handler is updated for v15)
- Updates to next-intl config after package updated and some things were deprecated
- Remove some unused dependencies and files
- Setup prettier-plugin-tailwindcss and run `lando npm run lint:fix` (**this causes many changed files**)
- Make contact info more generic
- Target ES6, not ES5
- Remove NEXT_IMAGE_DOMAIN (derive from CMS url instead)
- Cleanup and tighten eslint rules a bit, fix some violations
- Cleanup next config a bit, use .mjs and typecheck
- Cleanup relative imports
- Add /api/_ping endpoint (aliased to /health), often required to monitor our sites are up
- Add .env.local.example, for development/debugging env vars

## How to test:

1. Check [feature environment](https://feature-updates-etc-again-n53e.next-drupal-starterkit.dev.wdr.io) and [CircleCI](https://app.circleci.com/pipelines/github/wunderio/next-drupal-starterkit?branch=feature%2Fupdates-etc-again) looks good
2. Checkout this branch locally and check the usual setup/dev/build scripts work without issues
3. ddev should be tested too (I only tested with lando, v3.23.22)
